### PR TITLE
GH-927 Fixed GMT formatting issue

### PR DESCRIPTION
**Pull request by olegz**
Fixed DateFormat definition to include 'ISO 8601 time zone' designator (see https://docs.oracle.com/javase/7/docs/api/java/text/SimpleDateFormat.html)
Added constamt TimeZone to dateFormat
Added test

### CODE REVIEW

**Comment by pivotal-issuemaster** [comment]
@olegz Please sign the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-cloud/spring-cloud-stream&pullRequestId=929)!

[Click here](https://cla.pivotal.io/sync/spring?repositoryId=spring-cloud/spring-cloud-stream&pullRequestId=929) to manually synchronize the status of this Pull Request.

See the [FAQ](https://cla.pivotal.io/about) for frequently asked questions.

**Comment by pivotal-issuemaster** [comment]
@olegz Thank you for signing the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-cloud/spring-cloud-stream&pullRequestId=929)!

**Comment by olegz** [comment]
@viniciusccarvalho actually X adds Z at the end (not the number) when DF is set to use GMT timezone. There is test showing this. Do you see something different?

**Comment by olegz** [comment]
@viniciusccarvalho 
Just to clarify a bit. . . 
The quoted 'Z' was just that. In fact that was part of the issue where the date appeared to be in GMT but was actually not. With 'X' option you actually delegating to the Formatter to inject 'Z' (see code snippet below) if and only if the offset zone equals to 0 (which can only be IF your current timezone is GMT or you have written the date as GMT), thus verifying that your date was written in GMT format.
```
case PATTERN_ISO_ZONE:   // 'X'
            value = calendar.get(Calendar.ZONE_OFFSET)
                    + calendar.get(Calendar.DST_OFFSET);

            if (value == 0) {
                buffer.append('Z');
                break;
            }
```

File: spring-cloud-stream-metrics/src/test/java/org/springframework/cloud/stream/metrics/config/MetricJsonSerializerTests.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+		Serializer ser = new Serializer();	
+		ser.serialize(metric, jsonGenerator, null);
+		jsonGenerator.flush();
+		
+		JSONObject json = new JSONObject(writer.toString());
+		String serializedTimestamp = json.getString("timestamp");
+		assertEquals("2017-04-24T18:56:37.188Z", serializedTimestamp);
+	}
+	
+	private void injectTimestamp(Date date, Metric<Number> metric) throws Exception {
```
Comment (viniciusccarvalho) [inline spring-cloud-stream-metrics/src/test/java/org/springframework/cloud/stream/metrics/config/MetricJsonSerializerTests.java]: Metric has a constructor that accepts a Date object, there's no need for overriding access rules of the the class. Just use `Metric<Number> metric = new Metric("Hello",123,date);`

Comment (olegz) [inline spring-cloud-stream-metrics/src/test/java/org/springframework/cloud/stream/metrics/config/MetricJsonSerializerTests.java]: Fair enough, didn't realize that at the time, will fix

**Comment by viniciusccarvalho** [comment]
Squashed and merged via 8fc0cf1a0c1b8f8edcf542734048e59e42c74383

**Pull request closed without merging**